buildkite/bazel-bench/bazel_bench.py

```diff
@@ -86,23 +111,29 @@ def _get_bazel_commits(date, bazel_repo_path):
     ]
     command_output = subprocess.check_output(args, cwd=bazel_repo_path)
     decoded = command_output.decode("utf-8").split("\n")
+    full_list = [line.strip("'") for line in decoded if line]
 
-    return [line.strip("'") for line in decoded]
+    return _evenly_spaced_sample(full_list, max_commits)
 
 
-def _get_platforms(project_name):
+def _get_platforms(project_name, whitelist):
     """Get the platforms on which this project is run on BazelCI.
+    Filter the results with a whitelist & remove duplicates.
 
     Args:
       project_name: a string: the name of the project. e.g. "Bazel".
+      whitelist: a list of string denoting the whitelist of supported platforms.
 
     Returns:
       A list of string: the platforms for this project.
     """
     http_config = bazelci.DOWNSTREAM_PROJECTS_PRODUCTION[project_name]["http_config"]
     configs = bazelci.fetch_configs(http_config, None)
     tasks = configs["tasks"]
-    return list(map(lambda k: bazelci.get_platform_for_task(k, tasks[k]), tasks))
+    ci_platforms_for_project = list(
+        map(lambda k: bazelci.get_platform_for_task(k, tasks[k]), tasks))
+
+    return list(set(filter(lambda p: p in whitelist, ci_platforms_for_project)))
```

Comment:
Same here - note that this returns a set, but why would we have to convert it to a list?

```python
return set(p for p in ci_platforms_for_project if p in whitelist)
```

Reply:
Changed to list comprehension.
For the 2nd point, I convert it back to a list here since converting to `set` is just an implementation detail of "getting rid of duplicates". But yeah if it's always a set, we might as well use it as a set to enforce the no-dup property.